www/viewgame

```diff
@@ -2004,48 +1982,73 @@ $median_minutes=$median_hours_and_minutes['minutes'];
     // nearest 5 minutes when the game is longer than 1 hour.
     if (($estimated_hours >= 1) or ($estimated_minutes >= 1)) {
-            echo "<span class=details>";
-            echo "<details><summary>Estimated play time: ";
-            if ($estimated_hours >= 1) {
-                if ($estimated_hours == 1) {
-                    echo $estimated_hours . " hour ";
-                } else if ($estimated_hours > 1) {
-                    echo $estimated_hours . " hours ";
-                }
-                if ($estimated_minutes >= 1)
-                echo "and ";
-            }
-            if ($estimated_minutes == 1) {
-                echo $estimated_minutes . " minute ";
-            } else if ($estimated_minutes > 1) {
-                echo $estimated_minutes . " minutes ";
+        echo "<details><summary>Estimated play time: ";
+        if ($estimated_hours >= 1) {
+            if ($estimated_hours == 1) {
+                echo $estimated_hours . " hour ";
+            } else if ($estimated_hours > 1) {
+                echo $estimated_hours . " hours ";
             }
-            $s = (count($alltimes) === 1) ? "" : "s";
-            echo "(based on " . count($alltimes) . " vote$s)</span></summary><br>";
-            echo "Members voted for the following times for this game:";
-            echo "<ul>";
-            foreach ($alltimes as $t) {
-                $t = convertTimeToText($t);
-                echo "<li>$t</li>";
-            }
-            echo "</ul></details>";
+            if ($estimated_minutes >= 1)
+            echo "and ";
+        }
+        if ($estimated_minutes == 1) {
+            echo $estimated_minutes . " minute ";
+        } else if ($estimated_minutes > 1) {
+            echo $estimated_minutes . " minutes ";
         }
+        $s = (count($alltimes) === 1) ? "" : "s";
+        echo "(based on " . count($alltimes) . " vote$s)</span></summary><br>";
+        echo "Members voted for the following times for this game:";
+        echo "<ul>";
+        foreach ($alltimes as $t) {
+            $t = convertTimeToText($t);
+            echo "<li>$t</li>";
+        }
+        echo "</ul></details>";
+    }
+}
 
 //--------------------Code for Estimated Median Time Ends Here-----------------
 
+if ($ratingCntTot == 0 && $memberReviewCnt == 0) {
+    displayMedian($estimated_hours, $estimated_minutes, $alltimes);
+    echo "<img src=\"/img/blank.gif\" class=\"star0\" title=\"No Ratings\">
+       No reviews yet";
+    if ($embargoCnt == 0)
+        echo " - <a href=\"review?id=$id&userid=$curuser\">be the first</a>";
+} else if ($ratingAvg == 0 && $memberReviewCnt == 0) {
+    echo "<a href=\"viewgame?id=$id&ratings\" "
+        . "title=\"View all ratings and reviews for this game\">"
+        . "$ratingCntTot rating" . ($ratingCntTot != 1 ? "s" : "")
+        . "</a><br>";
+    displayMedian($estimated_hours, $estimated_minutes, $alltimes);
```

Comment:
This is a weird passage that probably can't happen in real life. I might delete it later in a separate PR.

The difference between `$ratingCntTot` and `$ratingCntAvg` is that you can publish a written review and check a box, "Omit my rating from the game's average." In that case, the rating is part of the "total" but it's not part of the "average," not part of the score at the top of the page.

It's possible for all of the ratings to be omitted, in which case, `$ratingAvg` will be null (which is `== 0`).

The first branch of this `if`/`else` chain considers the case where there are 0 ratings and 0 reviews. The third considers where there's at least one rating or at least one review, and handles the case where the `$ratingAvg` is null.

This middle section imagines a case where `$ratingAvg` is null, but `$memberReviewCnt` is 0. That's not something that can happen just by clicking around in the web UI, but it is representable in the database, if you write a review with an omitted rating, and then update the DB directly to null out the `review`.